src/sql/src/plan/statement/ddl.rs

```diff
@@ -1407,7 +1411,217 @@ pub fn plan_create_views(
                 materialize: materialized,
             }))
         }
-        CreateViewsDefinitions::Source { .. } => bail!("cannot create view from source"),
+        CreateViewsDefinitions::Source {
+            name: source_name,
+            targets,
+        } => {
+            let source_connector = normalize::unresolved_object_name(source_name.clone())
+                .map_err(anyhow::Error::new)
+                .and_then(|name| {
+                    scx.catalog
+                        .resolve_item(&name)
+                        .and_then(|item| item.source_connector())
+                        .map(|s| s.clone())
+                        .map_err(anyhow::Error::new)
+                });
+            match source_connector? {
```

Comment:
Using the imperative style is probably more idiomatic Rust here. (also avoids a clone)

```diff
diff --git a/src/sql/src/plan/statement/ddl.rs b/src/sql/src/plan/statement/ddl.rs
index 6f4753ee2..b318fbb50 100644
--- a/src/sql/src/plan/statement/ddl.rs
+++ b/src/sql/src/plan/statement/ddl.rs
@@ -1415,16 +1415,9 @@ pub fn plan_create_views(
             name: source_name,
             targets,
         } => {
-            let source_connector = normalize::unresolved_object_name(source_name.clone())
-                .map_err(anyhow::Error::new)
-                .and_then(|name| {
-                    scx.catalog
-                        .resolve_item(&name)
-                        .and_then(|item| item.source_connector())
-                        .map(|s| s.clone())
-                        .map_err(anyhow::Error::new)
-                });
-            match source_connector? {
+            let name = normalize::unresolved_object_name(source_name.clone())?;
+            let source_connector = scx.catalog.resolve_item(&name)?.source_connector()?;
+            match source_connector {
                 SourceConnector::External {
                     connector:
                         ExternalSourceConnector::Postgres(PostgresSourceConnector {
@@ -1486,12 +1479,12 @@ pub fn plan_create_views(
                     // An index from table_name -> schema_name -> PostgresTable
                     let mut details_info_idx: HashMap<String, HashMap<String, PostgresTable>> =
                         HashMap::new();
-                    for table in details.tables {
+                    for table in &details.tables {
                         details_info_idx
                             .entry(table.name.clone())
                             .or_default()
                             .entry(table.namespace.clone())
-                            .or_insert(table);
+                            .or_insert(table.clone());
                     }

                     for target in targets {
```

Reply:
What is the advantage to iterating `details.tables` through a reference and `clone()`ing each table versus just moving them into the hashmap? If I understand the rust docs correctly the for loop desugars into std::Vec::IntoIterator which does not clone as far as I can tell from the source and there is no further need for the source struct so emptying its vector seems fine.